Tests/Serializer/BaseSerializerHandlerTest.php

```diff
@@ -0,0 +1,144 @@
+
+/**
+ * @author Ahmet Akbana <ahmetakbana@gmail.com>
+ */
+final class BaseSerializerHandlerTest extends \PHPUnit_Framework_TestCase
+{
+    public function testGetSubscribingMethods()
+    {
+        $manager = $this->getMock('Sonata\CoreBundle\Model\ManagerInterface');
+
+        $serializer = new FooSerializer($manager);
+
+        $expectedMethods = array(
+            array(
+                'direction' => GraphNavigator::DIRECTION_SERIALIZATION,
+                'format' => 'json',
+                'type' => 'foo',
+                'method' => 'serializeObjectToId',
+            ),
+            array(
+                'direction' => GraphNavigator::DIRECTION_DESERIALIZATION,
+                'format' => 'json',
+                'type' => 'foo',
+                'method' => 'deserializeObjectFromId',
+            ),
+            array(
+                'direction' => GraphNavigator::DIRECTION_SERIALIZATION,
+                'format' => 'xml',
+                'type' => 'foo',
+                'method' => 'serializeObjectToId',
+            ),
+            array(
+                'direction' => GraphNavigator::DIRECTION_DESERIALIZATION,
+                'format' => 'xml',
+                'type' => 'foo',
+                'method' => 'deserializeObjectFromId',
+            ),
+            array(
+                'direction' => GraphNavigator::DIRECTION_SERIALIZATION,
+                'format' => 'yml',
+                'type' => 'foo',
+                'method' => 'serializeObjectToId',
+            ),
+            array(
+                'direction' => GraphNavigator::DIRECTION_DESERIALIZATION,
+                'format' => 'yml',
+                'type' => 'foo',
+                'method' => 'deserializeObjectFromId',
+            ),
+        );
+
+        $methods = $serializer->getSubscribingMethods();
+
+        $this->assertSame($methods, $expectedMethods);
+    }
+
+    public function testSerializeObjectToIdWithDataIsInstanceOfManager()
+    {
+        $data = $this->getMock(
+            'Sonata\CoreBundle\Tests\Fixtures\Bundle\Serializer\FooSerializer',
+            array('getId'),
+            array(),
+            '',
+            false
+        );
+
+        $data->expects($this->once())
+            ->method('getId')
+            ->willReturn(1);
+
+        $manager = $this->getMock('Sonata\CoreBundle\Model\ManagerInterface');
+        $manager->expects($this->once())
+            ->method('getClass')
+            ->willReturn(get_class($data));
+
+        $context = $this->getMock('JMS\Serializer\Context');
+
+        $visitor = $this->getMock('JMS\Serializer\VisitorInterface');
+        $visitor->expects($this->once())
+            ->method('visitInteger')
+            ->with(1, array('foo'), $context)
+            ->willReturn(true);
+
+        $serializer = new FooSerializer($manager);
+
+        $this->assertTrue($serializer->serializeObjectToId($visitor, $data, array('foo'), $context));
```

Comment:
I am manuplating the return for `getClass` method of `$manager` to an instance of my fake class `FooSerializer`. And I set the variable `$data` to instance of my fake class as well. 'Data is Instance of Manager' is the logic in the method but test does not work with exact instance of manager of course.